sys/mips/mips/trap.c

```diff
@@ -1088,13 +1098,17 @@ trap(struct trapframe *trapframe)
 	case T_IWATCH + T_USER:
 	case T_DWATCH + T_USER:
 		{
-			intptr_t va;
+			char * __CAPABILITY va;
 
 			/* compute address of trapped instruction */
+#ifdef CPU_CHERI
+			va = trapframe->pcc;
+#else
 			va = trapframe->pc;
+#endif
 			if (DELAYBRANCH(trapframe->cause))
 				va += sizeof(int);
-			printf("watch exception @ %p\n", (void *)va);
+			printf("watch exception @ %p\n", (__cheri_fromcap void *)va);
```

Comment:
I think this is right for now, but will eventually break when kernel DDC doesn't cover all address space.

Reply:
Hmm, no this is fine even once DDC changes.  I was thinking of the conversion the other way.